### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by burke**

### CODE REVIEW

File: examples/minimal/stargate.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift("/opt/dev/vendor/deps/cli-kit/lib")
```
Comment (lugray) [inline examples/minimal/stargate.rb]: I'm not sure this location makes sense for public consumption.

File: examples/minimal/stargate.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+$LOAD_PATH.unshift("/opt/dev/vendor/deps/cli-ui/lib")
+
+require 'cli/ui'
+require 'cli/kit'
+
+CLI::UI::StdoutRouter.enable
+
+module Stargate
```
Comment (lugray) [inline examples/minimal/stargate.rb]: I'd maybe call this `Example` instead of `Stargate`

File: examples/minimal/stargate.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  Registry = CLI::Kit::CommandRegistry.new(default: 'help', contextual_resolver: nil)
+  Registry.add(Class.new(CLI::Kit::BaseCommand) do
+    def call(_args, _name)
+      puts "submit"
+    end
+
+    def self.help
+      "Submit a stargate job.\nUsage: {{command:stargate} submit}}"
+    end
+  end, 'submit')
```
Comment (lugray) [inline examples/minimal/stargate.rb]: Maybe `demo`?

File: examples/minimal/stargate.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  EntryPoint   = ->(args) { Executor.call(*Resolver.call(args)) }
+
+  Registry = CLI::Kit::CommandRegistry.new(default: 'help', contextual_resolver: nil)
+  Registry.add(Class.new(CLI::Kit::BaseCommand) do
+    def call(_args, _name)
+      puts "submit"
+    end
+
+    def self.help
+      "Submit a stargate job.\nUsage: {{command:stargate} submit}}"
```
Comment (lugray) [inline examples/minimal/stargate.rb]: What it actually does (Print some example output)

**Comment by burke** [comment]
Revised, and added `examples/README.md`

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Examples
+
+**Note: If you're looking to bootstrap a new app, consider running `cli-kit new` instead of
+copy/pasting one of these.**
+
+# Full Example
```
Comment (lavoiesl) [inline examples/README.md]: Should use `##` as a second-level header.

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+# Single-File starting point
+
+Maybe it appeals to you to start from a single file and grow, rather than generating framework.
+Check out the `examples/single-file` example.
+
+# Minimal example
+
+This example demonstrates the core of what `cli-kit` does very simply, but in general, shouldn't be
+used for anything other than quick hacks. It leaves out a few best practices that help CLI apps
+scale gracefully. Find it at `examples/minimal`.
```
Comment (lavoiesl) [inline examples/README.md]: You can use this to deep-link, relative to the README:
```md
[`examples/minimal`](minimal)
```

File: examples/single-file/example.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+  autocall(:ErrorHandler) do
+    CLI::Kit::ErrorHandler.new(
+      log_file: LOG_FILE,
+      exception_reporter: nil
+    )
+  end
+end
+
+exit(Example::ErrorHandler.call do
```
Comment (lavoiesl) [inline examples/single-file/example.rb]: If feel it'd be easier to read with a block `if`

**Comment by burke** [comment]
@lugray I substantially simplified the minimal example even further. I think having both is better-justified now.

File: examples/single-file/example.rb [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+      default: 'hello',
+      contextual_resolver: nil
+    )
+
+    def self.register(const, cmd, path = nil, &block)
+      path ? autoload(const, path) : autocall(const, &block)
+      Registry.add(->() { const_get(const) }, cmd)
+    end
+
+    # register(:Submit, 'submit', 'a/b/hello')
```
Comment (lugray) [inline examples/single-file/example.rb]: `# register(:Hello, 'hello', 'a/b/hello')`

**Pull request merged**